.github/workflows/ipt.yml

```
@@ -0,0 +1,65 @@
+name: IPT Build/Release
+
+on:
+  push:
+    # takes muliple branch names
+    branches: 
+      - main
+      - '[0-9][0-9][0-9][0-9].[0-9][0-9].*' # 2021.01.x
+    tags:
+      - '[0-9][0-9][0-9][0-9].[0-9][0-9].[0-9][0-9]' # 2021.01.01
+ 
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version'
+        required: true
+        default: 'YYYY.MINOR.MICRO'
+env:
+  REPO_URL: ${{ github.repository }}
+    
+jobs:
+  build:
+    # to test a feature, change the repo name to your github id
+    if: github.repository_owner == 'tl-its-umich-edu' || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    steps:
+  
+      - name: Checkout code 
+        uses: actions/checkout@v4
+    
+      - name: Extract branch name
+        id: extract_branch
+        run: echo "BRANCH_NAME=$(basename ${{ github.ref }})" >> $GITHUB_ENV
+      
+      - name: build Docker image
+        run: |
+          docker build -f dockerfiles/Dockerfile.prod . --tag ghcr.io/${{ env.REPO_URL }}:${BRANCH_NAME}
+      
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+            registry: ghcr.io
+            username: ${{ github.actor }}
+            password: ${{ secrets.GITHUB_TOKEN }}
+      
+      - name: Push Docker image to GitHub Container Registry
+        run: |
+          docker push ghcr.io/${{ env.REPO_URL }}:${BRANCH_NAME}
+    
+  release: 
+     # Making sure that release only runs for tag pushes
+    if: startsWith(github.ref, 'refs/tags/') && github.repository_owner == 'tl-its-umich-edu'
```

This check for repository_owner doesn't need to be in this step.
```suggestion
    if: startsWith(github.ref, 'refs/tags/')'
```